examples/model_evaluation/plot_feature_importance.py

```diff
@@ -0,0 +1,551 @@
+# In particular, we perform a K-means clustering on the geospatial features:
+
+# %%
+from sklearn.cluster import KMeans
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import PolynomialFeatures, SplineTransformer
+
+geo_columns = ["Latitude", "Longitude"]
+
+preprocessor = make_column_transformer(
+    (KMeans(n_clusters=10), geo_columns),
+    remainder="passthrough",
+)
+engineered_ridge = make_pipeline(
+    preprocessor,
+    SplineTransformer(),
+    PolynomialFeatures(degree=2, interaction_only=True),
+    Ridge(),
+)
+engineered_ridge
+
+# %%
+# Now, let us compute the metrics and compare it to our previous model using
+# a :class:`skore.ComparisonReport`:
+
+# %%
+from skore import ComparisonReport
+
+engineered_ridge_report = EstimatorReport(
+    engineered_ridge,
+    X_train=X_train,
+    X_test=X_test,
+    y_train=y_train,
+    y_test=y_test,
+)
+comparator = ComparisonReport(reports=[ridge_report, engineered_ridge_report])
+comparator.metrics.report_metrics()
+
+# %%
+# We get a much better score!
+# However, interpreting the features is harder: indeed, our complex feature engineering
+# introduced a lot of features:
+
+# %%
+print("Initial number of features:", X_train.shape[1])
+
+X_train_transformed = engineered_ridge_report.estimator_[:-1].transform(X_train)
+n_features_engineered = X_train_transformed.shape[1]
+print("Number of features after feature engineering:", n_features_engineered)
+
+# %%
+# Let us display the 15 largest absolute coefficients:
+
+# %%
+df_engineered_ridge_coef = engineered_ridge_report.feature_importance.coefficients()
+
+
+def sort_absolute_values(df):
+    df = df.assign(absolute_coefficient=df["Coefficient"].abs()).sort_values(
+        by="absolute_coefficient", ascending=False
+    )
+    return df.head(15)
+
+
+sort_absolute_values(df_engineered_ridge_coef)
+
+# %%
+# We can observe that the most importance features are interactions between several
+# features, that a simple linear model without feature engineering could not have
+# captured.
+
+# %%
+# Compromising on complexity
+# --------------------------
+
+# %%
+# Now, let us build a model with a more interpretable feature engineering, although
+# it might not perform as well.
+# For that, after the complex feature engineering, we perform some feature selection
+# using a :class:`~sklearn.feature_selection.SelectKBest`.
+# To compensate the drop in score, we fine-tune some hyperparameters using
+# a :class:`~sklearn.model_selection.RandomizedSearchCV`.
+
+# %%
+from scipy.stats import randint
+from sklearn.feature_selection import SelectKBest, VarianceThreshold
+from sklearn.model_selection import RandomizedSearchCV
+
+preprocessor = make_column_transformer(
+    (KMeans(n_clusters=20), geo_columns),
+    remainder=SplineTransformer(),
+)
+model = make_pipeline(
+    preprocessor,
+    PolynomialFeatures(degree=1, interaction_only=True),
+    VarianceThreshold(),
+    SelectKBest(k=50),
+    Ridge(),
```

Comment:
Let's do a `RidgeCV`:


```suggestion
    RidgeCV(np.logspace(-5, 5, num=100)),
```

Because it will use a GCV under the hood, it will be much cheaper than doing it in the search.
The small drawback is that we have a small data leak but we can discard it.

Reply:
I did just Ridge instead of RidgeCV so that it's easy for me to do the parallel coordinates plot below?

Reply:
`RidgeCV` will expose an `alpha_` parameter. I also commented that the parallel coordinate plot is maybe not necessary right now.